driver/SDL/SDL_keyboard.c

```diff
@@ -0,0 +1,150 @@
+/**
+ * @file keyboard.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "SDL_keyboard.h"
+#if USE_KEYBOARD != 0
+
+/*********************
+ *      DEFINES
+ *********************/
+#ifndef MONITOR_ZOOM
+#define MONITOR_ZOOM    1
+#endif
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+static bool keyup = false;
+static bool keydown = false;
+static bool repeat = false;
+static bool keyboard_was_read = true;
+static int last_key = -1;
+
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+/**
+ * Get the current key and state of the keyboard
+ * @param data store the keyboard data here
+ * @return false
+ */
+void keyboard_read(struct _lv_indev_drv_t * indev_drv, lv_indev_data_t * data)
+{
+    if(!keyboard_was_read)
+    {
+        /* Store the collected data */
+        if(keyup)
+        {
+            data->state = LV_INDEV_STATE_PR;
+            data->key = last_key;
+        }
+        else if(keydown)
+        {
+            data->state = LV_INDEV_STATE_REL;
+            data->key = last_key;
+        }
+    }
+    if(keydown && repeat){
+        data->state = LV_INDEV_STATE_PR;
+        data->key = last_key;
+    }
+    
+    data->continue_reading = false;
+    keyboard_was_read = true;
+    return;
+}
+
+/**
+ * It will be called from the main SDL thread
+ */
+void keyboard_handler(SDL_Event * event)
+{
+    keydown = false;
+    keyup = false;
+    last_key = -1;
+
+    if(keyboard_was_read)
```

Comment:
Keys are recorded only when `keyboard_was_read` is true. `keyboard_was_read` is set to true only after `keyboard_read` was called by LVGL, and set to false by `keyboard_handler`.  

Suppose `keyboard_handler` is called more frequently than `keyboard_read`. This means that keys are recorded on the **first** invocation of `keyboard_handler` after a call to `keyboard_read` and not updated until the next call to `keyboard_read`. If keyboard state changes, that would not be recorded (until the next call to `keyboard_read` followed by a call to `keyboard_handler`).  

Is it designed that way on purpose?   
Perhaps it would make more sense to record the keyboard state on the **last** invocation of `keyboard_handler` before `keyboard_read` is called?